app/src/main/java/org/stepik/android/adaptive/pdd/util/ExpUtil.java

```diff
@@ -1,22 +1,43 @@
 package org.stepik.android.adaptive.pdd.util;
 
+import org.stepik.android.adaptive.pdd.api.API;
 import org.stepik.android.adaptive.pdd.data.AnalyticMgr;
 import org.stepik.android.adaptive.pdd.data.SharedPreferenceMgr;
+import org.stepik.android.adaptive.pdd.data.db.DataBaseMgr;
+
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
 
 public class ExpUtil {
     private static String EXP_KEY = "exp_key";
     private static String STREAK_KEY = "streak_key";
 
+    private static CompositeDisposable compositeDisposable = new CompositeDisposable();
+
     public static long getExp() {
         return SharedPreferenceMgr.getInstance().getLong(EXP_KEY);
     }
 
-    public static long changeExp(long delta) {
+    public static long changeExp(long delta, long submissionId) {
         long exp = SharedPreferenceMgr.getInstance().changeLong(EXP_KEY, delta);
         AnalyticMgr.getInstance().onExpReached(exp - delta, delta);
+
+        compositeDisposable.add(
+                Completable
+                        .fromRunnable(() -> DataBaseMgr.getInstance().onExpGained(delta, submissionId))
+                        .andThen(syncRating())
+                        .subscribeOn(Schedulers.io())
+                        .subscribe((__) -> {}, (e) -> AnalyticMgr.getInstance().onRatingError()));
```

Comment:
If Internet connection is lost, then we will get onRatingError, is it OK? maybe we should get this action only on 5XX responses?